spec/core/ics-004-channel-and-packet-semantics/UPGRADES.md

```diff
@@ -209,13 +215,13 @@ At the end of an upgrade handshake between two chains implementing the sub-proto
 - Each chain is running their new upgraded channel end and is processing upgraded logic and state according to the upgraded parameters.
 - Each chain has knowledge of and has agreed to the counterparty's upgraded channel parameters.
 
-If a chain does not agree to the proposed counterparty `UpgradedChannel`, it may abort the upgrade handshake by writing an error receipt into the `errorPath` and restoring the original channel. The error receipt MAY be arbitrary bytes and MUST be non-empty.
+If a chain does not agree to the proposed counterparty upgrade `ChannelEnd`, it may abort the upgrade handshake by writing an error receipt into the `channelUpgradeErrorPath` and restoring the original channel. The upgrade error receipt MAY be arbitrary bytes and MUST be non-empty.
 
-`errorPath(id) => error_receipt`
+`channelUpgradeErrorPath(id) => upgrade_error_receipt`
 
-A relayer may then submit a `ChanUpgradeCancelMsg` to the counterparty. Upon receiving this message a chain must verify that the counterparty wrote a non-empty error receipt into its `UpgradeError` and if successful, it will restore its original channel as well thus cancelling the upgrade.
+A relayer may then submit a `CancelChannelUpgradeMsg` to the counterparty. Upon receiving this message a chain must verify that the counterparty wrote a non-empty upgrade error receipt into its `channelUpgradeErrorPath` and if successful, it will restore its original channel as well thus cancelling the upgrade.
```

Comment:
`CancelChannelUpgradeMsg` and `ChannelUpgradeTimeoutMsg` should have the same format. Either `CancelChannelUpgradeMsg` and `TimeoutChannelUpgradeMsg` or `ChannelUpgradeCancelMsg` and `ChannelUpgradeTimeoutMsg`. I prefer the latter.

Reply:
All right, if you prefer `ChannelUpgradeCancelMsg` and `ChannelUpgradeTimeoutMsg`, then should we always put the verb just before `Msg`? If we want to be consistent across specs, then that would require to [change `RegisterCounterpartyAddressMsg`](https://github.com/cosmos/ibc/tree/master/spec/app/ics-029-fee-payment#user-interaction-with-fee-middleware) into `CounterpartyAddressRegisterMsg`, and also [`CancelConnectionUpgradeMsg` and `CancelConnectionUpgradeTimeoutMsg`](https://github.com/cosmos/ibc/blob/master/spec/core/ics-003-connection-semantics/UPGRADES.md#sub-protocols), although `ConnectionUpgradeTimeoutCancelMsg` doesn't sound like music to my ears, which makes me think that maybe putting the verb at the beginning would be better... What do you think? 

Depending on what we decide I can change the others in a separate PR.

Reply:
I think it will be difficult to keep this naming consistency across all applications, but yeah, that would be nice. In general though, I think we should aim for the naming to be consistent across IBC core specs. 

Regarding `ConnectionUpgradeTimeoutCancelMsg`, I'm not sure why the name is like that. I think it should be `ConnectionUpgradeTimeoutMsg` (see the equivalent in [ICS 4](https://github.com/cosmos/ibc/blob/master/spec/core/ics-004-channel-and-packet-semantics/UPGRADES.md#upgrade-handshake)).

Reply:
Ah, true. Must be a typo. Haven't read the connection upgrade spec fully yet. I can open a PR later on.